nano/node/messages.cpp

```diff
@@ -707,6 +707,21 @@ bool nano::keepalive::deserialize (nano::stream & stream_a)
 	return error;
 }
 
+std::string nano::keepalive::to_string () const
+{
+	std::stringstream stream;
+
+	stream << header.to_string ();
+
+	for (auto peer = peers.begin (), peers_end = peers.end (); peer != peers_end; ++peer)
```

Comment:
There is no need to take a copy of `peers_end = peers.end ();` to avoid calling the function repeatedly. compilers know how to optimise these things easily. The code should be written for readability first, not for optimisation.

This is better because it is easily readable and because the compiler will make the optimisation anyway:
```
for (auto peer = peers.begin (); peer != peers.end(); ++peer)
```